Classes/Domain/Model/Changes/MoveAfter.php

```diff
@@ -71,19 +75,49 @@ public function apply(): void
             $contentRepository = $this->contentRepositoryRegistry->get($subject->contentRepositoryId);
 
-            $command = MoveNodeAggregate::create(
-                $subject->workspaceName,
-                $subject->dimensionSpacePoint,
-                $subject->aggregateId,
-                RelationDistributionStrategy::STRATEGY_GATHER_ALL,
-                $hasEqualParentNode ? null : $parentNodeOfPreviousSibling->aggregateId,
-                $precedingSibling->aggregateId,
-                $succeedingSibling?->aggregateId,
-            );
-            $contentRepository->handle($command);
+            if (!$precedingSibling->dimensionSpacePoint->equals($subject->dimensionSpacePoint)) {
+                // WORKAROUND for MOVE ACROSS DIMENSIONS:
+                // - we want it to work like a copy/paste, followed by an original delete.
+                // - This is to ensure the user can use it as expected from text editors, where context
+                //   is not preserved during cut/paste.
+                // - LATERON, we need to expose CreateNodeVariant (which creates connected variants) in the UI as well.
+                $command = CopyNodesRecursively::createFromSubgraphAndStartNode(
+                    $contentRepository->getContentGraph($subject->workspaceName)->getSubgraph(
+                        $subject->dimensionSpacePoint,
+                        VisibilityConstraints::withoutRestrictions()
+                    ),
+                    $subject->workspaceName,
+                    $subject,
+                    // NOTE: in order to be able to copy/paste across dimensions, we need to use
+                    // the TARGET NODE's DimensionSpacePoint to create the node in the target dimension.
+                    OriginDimensionSpacePoint::fromDimensionSpacePoint($precedingSibling->dimensionSpacePoint),
+                    $parentNodeOfPreviousSibling->aggregateId,
+                    $succeedingSibling?->aggregateId
+                );
+
+                $contentRepository->handle($command);
+
+                $command = RemoveNodeAggregate::create(
+                    $subject->workspaceName,
+                    $subject->aggregateId,
+                    $subject->dimensionSpacePoint,
+                    NodeVariantSelectionStrategy::STRATEGY_ALL_SPECIALIZATIONS,
+                );
+                $contentRepository->handle($command);
+            } else {
+                $command = MoveNodeAggregate::create(
+                    $subject->workspaceName,
+                    $subject->dimensionSpacePoint,
+                    $subject->aggregateId,
+                    RelationDistributionStrategy::STRATEGY_GATHER_ALL,
+                    $hasEqualParentNode ? null : $parentNodeOfPreviousSibling->aggregateId,
+                    $precedingSibling->aggregateId,
+                    $succeedingSibling?->aggregateId,
+                );
+                $contentRepository->handle($command);
+            }
```

Comment:
im not sure i understand the full implications of this ... as far as i know is move node super well tested and supports a variety of cases to distribute node aggregates and i dont know if we want to trade that in for the use of this `CopyNodesRecursively` plus removing the original node?

isnt this supported already by `MoveNodeAggregate` or do we need to extend the core?